src/components/app.js

```diff
@@ -42,9 +43,13 @@ class App extends React.Component<ConnectedProps & OwnProps> {
           <Router history={history}>
             <Switch>
               <Route exact path="/" component={HomePage} />
-              <Route exact path="/issues" component={IssuesPage} />
-              <Route path="/issues/:issueId" component={IssueDetails} />
               <Route path="/login" component={Login} />
+              <Route exact path="/:name/:repo/issues" component={IssuesPage} />
+              <Route
+                path="/:name/:repo/issues/:number"
+                component={IssueDetails}
+              />
+              <Route exact path="/404" component={Error404} />
```

Comment:
use a generic name for your error page and display relevant message according to error code